test/automatic_differentiation_test.jl

```diff
@@ -1,52 +1,52 @@
 using DualNumbers
 using Base.Test
 
-x = dual(2, 1)
+x = Dual(2, 1)
 y = x^3
 
-@test_approx_eq real(y) 2.0^3
-@test_approx_eq epsilon(y) 3.0*2^2
+@test_approx_eq value(y) 2.0^3
+@test_approx_eq dual(y) 3.0*2^2
 
 y = x^3.0
 
-@test_approx_eq real(y) 2.0^3
-@test_approx_eq epsilon(y) 3.0*2^2
+@test_approx_eq value(y) 2.0^3
+@test_approx_eq dual(y) 3.0*2^2
 
 y = sin(x)+exp(x)
-@test_approx_eq real(y) sin(2)+exp(2)
-@test_approx_eq epsilon(y) cos(2)+exp(2)
+@test_approx_eq value(y) sin(2)+exp(2)
+@test_approx_eq dual(y) cos(2)+exp(2)
 
 @test x > 1
 y = abs(-x)
-@test_approx_eq real(y) 2.0
-@test_approx_eq epsilon(y) 1.0
+@test_approx_eq value(y) 2.0
+@test_approx_eq dual(y) 1.0
 
-@test isequal(1.0,dual(1.0))
+@test isequal(1.0,Dual(1.0))
 
 y = 1/x
-@test_approx_eq real(y) 1/2
-@test_approx_eq epsilon(y) -1/2^2
+@test_approx_eq value(y) 1/2
+@test_approx_eq dual(y) -1/2^2
 
 Q = [1.0 0.1; 0.1 1.0]
-x = dual([1.0,2.0])
-x[1] = dual(1.0,1.0)
+x = Dual([1.0,2.0])
+x[1] = Dual(1.0,1.0)
 y = (1/2)*dot(x,Q*x)
-@test_approx_eq real(y) 2.7
-@test_approx_eq epsilon(y) 1.2
+@test_approx_eq value(y) 2.7
+@test_approx_eq dual(y) 1.2
 
 function squareroot(x)
     it = x
     while abs(it*it - x) > 1e-13
-        it = it - (it*it-x)/(2it)
+        it = (it+x/it)/2
```

Comment:
It's just two fewer operations https://en.wikipedia.org/wiki/Square_root#Computation